packages/docs/community/release-process/smart-contracts.md

```diff
@@ -37,57 +37,71 @@ When a release is getting proposed on a Baklava/Alfajores/Mainnet, the commit th
 
 
-## Build and Release Process
+## Release Process
 
-Using the following scripts, new contracts can be built and deployed alongside a corresponding on-chain governance proposal. Run the following scripts from the branch with the new release contracts:
+Using the following procedure, a new contract release can be built, deployed, and proposed for upgrade automatically.
 
+### Verify Current Deployment
+
+Compile the contracts in <code>$RELEASE_BRANCH</code> branch and verify that the deployed bytecode on <code>$NETWORK</code> matches compiled bytecode.
 ```bash
-## Script 1 ##
 # TODO: command to pull source code from release N-1, compile and get the artifacts
-# verify the current contract bytecode on the network matches the contracts currently on the RC1 branch
-yarn truffle exec --network $NETWORK ./scripts/truffle/verify-bytecode.js --build_artifacts build/rc1 --before_release_1
+yarn verify-deployed -n $NETWORK -b $RELEASE_BRANCH
+```
 
-## Script 2 ##
-# TODO: command to pull source code from current release branch
-# generate report on backwards compatibility between RC1 and current branch with new release contracts
-yarn ts-node scripts/check-backward.ts report --exclude ".*Test|Mock.*|I[A-Z].*|.*Proxy|LinkedList|SortedLinkedList|SortedLinkedListWithMedian|MultiSig.*|ReleaseGold" -o build/rc1 -n build/contracts --output_file report.json
+### Check Backward Compatibility
+
+Compile the contracts in <code>$RELEASE_BRANCH</code> and <code>$BRANCH</code> branches and check backwards compatibility against a release, subject to the following exceptions:
+  1. If the STORAGE version has changed, do not perform backwards compatibility checks
+  2. If the MAJOR version has changed, check storage layout is backwards compatible, but do not check contract ABI is backwards compatible
 
-## Script 3 ##
-# deploy new contracts and generate governance proposal for upgrade
-yarn truffle exec --network development ./scripts/truffle/make-release.js --build_directory build/ --report report.json --proposal proposal.json --initialize_data example-initialize-data.json'
+For changed contracts, confirm that version number in <code>$BRANCH</code> is strictly greater than <code>$RELEASE_BRANCH</code> version number.
+For unchanged contracts, confirm that version number in <code>$BRANCH</code> is exactly the <code>$RELEASE_BRANCH</code> version number.
+Output a JSON backwards compatibility report to <code>$REPORT</code> file.
+
+```bash
+# TODO: command to pull source code from current release branch
+yarn check-versions -a $RELEASE_BRANCH -b $BRANCH -r $REPORT
 ```
 
-These scripts do the following:
+### Deploy New Contracts
+
+Build contracts on <code>$BRANCH</code> and, using corresponding backwards compatibility report <code>$REPORT</code>, deploy changed contracts to <code>$NETWORK</code>.
+STORAGE updates are adopted by deploying a new proxy/implementation pair. Initialize new contracts with values using <code>$INITIALIZE_DATA</code> when specified.
```

Comment:
What is `$INITIALIZE_DATA`?

Reply:
for initializing storage incompatible contracts which are being deployed anew (ie `DowntimeSlasher`)

Reply:
Where does it come from? Should we have a process for this?